### ISSUE TITLE
Differences in develop and master branch.

### ISSUE THREAD

**Issue by briansp2020** [HSA-Libraries/Bolt#64]
I noticed that there are some differences between develop branch and master/v1.0 branch. Is that intentional? Most of the differences are minor documentation differences that probably won't affect much. But it seems like something was missed when syncing v1.0/master/develop branches in preparation of v1.0 release.

I'm just getting started with Git way of doing thing. I've only been using SVN. So, it's possible that I'm missing something...

Also, to submit an entry for Bolt Sample Code Contest, I should open a pull request to develop branch, right? Or does it not matter?

**Comment by jayavanth** [comment]
Hi Brian,
The difference between develop and master/v1.0 branches are intentional. With this release in context, develop branch contains code for future releases, master branch contains code for the latest release and v1.0 branch is for the current release.

Ideally, your samples development which will be in Examples folder should be independent of the Bolt library. So, it doesn't matter if you're working on either develop or master branch. Nevertheless, your pull requests will be accepted only for develop branch.

In conclusion, you can fork the repo, commit your code and send a pull request to us. We'll review the code and then you can make fixes if any.

 -- Jay

**Comment by kknox** [comment]
You can read more about the [branching and development model](http://nvie.com/posts/a-successful-git-branching-model/) that Bolt has adopted.

**Comment by briansp2020** [comment]
Kent,
Thanks for pointing me to the article. I think I understand the development model better now. However, there still seems to be some discrepancies that does not quite make sense to me.

For example, in develop branch, "doxy/mainpage.dox" line 203 says "Bolt function can be executed" where as the v1.0/master says "Bolt function is designed to be executed". I see some mismatches in line 227, 229 & 210 as well. I also see similar problem in SupportedFunction.dox line 23 to 25. It looks to me like v1.0 was not merged to develop branch properly since those were updated specifically for v1.0 release. README.md (line 14,15) seems odd as well. I can see that commit f466982de63ed187265db38dfe744f5704f0b7f5 and 8229e7bc066a8678adbb268f54106014c831e4bd was not intended to be released in v1.0.

Also, "include\bolt\cl\device_vector.h" line 717 was updated as part of commit 8229e7bc066a8678adbb268f54106014c831e4bd. That also looks like a mistake since it does not seem related to "Added a new python script to run the new benchmark code." which is the description of the commit.

Sorry to keep bothering you about closed issue/question. I'm still learning and hoping to get up to speed soon so that I can make some real useful contribution (and win a note book :))

![image](https://f.cloud.github.com/assets/3746601/779631/20aa86f8-e9e2-11e2-9ce1-a9ac6739a174.png)

**Comment by ravibanger** [comment]
Thanks Brian for good look at the different branches in BOLT. Yes you rightly recognized that develop branch was not merged with v1.0/master. I will be doing that soon, just giving it some time (a day more). The develop branch will have some code/fixes added on daily basis. 
For your information you can develop by branching out of master or develop, (preferably master because it is a tested code and will not be modified in the longer run). and your pull requests can be merged to the develop branch, once we complete the review. 
Wish you good luck in the competition.

Regards,
Banger.
